backend/controller/observability/cron.go

```diff
@@ -0,0 +1,106 @@
+package observability
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+
+	"github.com/TBD54566975/ftl/internal/model"
+	"github.com/TBD54566975/ftl/internal/observability"
+)
+
+const (
+	cronMeterName       = "ftl.cron"
+	cronJobRefAttribute = "ftl.cron.job.ref"
+)
+
+type CronMetrics struct {
+	jobFailures metric.Int64Counter
+	jobsKilled  metric.Int64Counter
+	jobsActive  metric.Int64UpDownCounter
+	jobLatency  metric.Int64Histogram
+}
+
+func initCronMetrics() (*CronMetrics, error) {
+	result := &CronMetrics{}
+
+	var errs error
+	var err error
+
+	meter := otel.Meter(deploymentMeterName)
+
+	counter := fmt.Sprintf("%s.job.failures", cronMeterName)
+	if result.jobFailures, err = meter.Int64Counter(
+		counter,
+		metric.WithDescription("the number of failures encountered while performing activities associated with starting or ending a cron job")); err != nil {
+		result.jobFailures, errs = handleInt64CounterError(counter, err, errs)
+	}
+
+	counter = fmt.Sprintf("%s.jobs.kills", cronMeterName)
+	if result.jobsKilled, err = meter.Int64Counter(
+		counter,
+		metric.WithDescription("the number cron jobs killed by the controller")); err != nil {
+		result.jobsKilled, errs = handleInt64CounterError(counter, err, errs)
+	}
+
+	counter = fmt.Sprintf("%s.jobs.active", cronMeterName)
+	if result.jobsActive, err = meter.Int64UpDownCounter(
+		counter,
+		metric.WithDescription("the number of actively executing cron jobs")); err != nil {
+		result.jobsActive, errs = handleInt64UpDownCounterError(counter, err, errs)
+	}
+
+	counter = fmt.Sprintf("%s.job.latency", cronMeterName)
+	if result.jobLatency, err = meter.Int64Histogram(
+		counter,
+		metric.WithDescription("the latency between the scheduled execution time and completion of a cron job"),
+		metric.WithUnit("ms")); err != nil {
+		result.jobLatency, errs = handleInt64HistogramCounterError(counter, err, errs)
+	}
+
+	return result, errs
+}
+
+func (m *CronMetrics) JobExecutionStarted(ctx context.Context, job model.CronJobKey, deployment model.DeploymentKey) {
+	m.jobsActive.Add(ctx, 1, metric.WithAttributes(
+		attribute.String(observability.ModuleNameAttribute, job.Payload.Module),
+		attribute.String(cronJobRefAttribute, job.String()),
```

Comment:
Just confirming - is `job.String()` the ref to the cron job declaration?

Reply:
It's the module qualified verb name

Reply:
Ahhh in that case should we rename the attribute key to match?

Reply:
updated